config.env.txt
# REQUIRED CONFIG
BOT_TOKEN = "1977831577:AAHVq-heDQVJiDQ8Ee6MqxtZ5JSVVMT0MIE" # Enter your telegram bot token
OWNER_ID = 1606221784 # Enter your telegram user id
TELEGRAM_API = 3975570 # Enter telegram API key
TELEGRAM_HASH = "680b62f2844aa1954216f6cb99d2f3d9" # Enter telegram API hash

# OPTIONAL CONFIG
AUTHORIZED_CHATS = "" # Enter telegram chat id to authorize it
SUDO_USERS = 5623055023 # Enter telegram user id for sudo access

DATABASE_URL = "mongodb+srv://ghost:ghost123@hunter.z1x1x.mongodb.net/?retryWrites=true&w=majority&appName=Hunter" # Enter mongo database URL for python 3.12 or higher
USER_SESSION_STRING "" # Check readme to learn how to generate this

# Optional API
FILELION_API = ""
STREAMWISH_API = ""

# GDrive Tools
GDRIVE_ID = "1-UhbfAlDYGulzPi6UlpVT7pVaZzAutGs"
INDEX_URL = "https://tghthingleech.theghosthunter-tg1.workers.dev/0:"
IS_TEAM_DRIVE = "True" # True or False
STOP_DUPLICATE = "True" # True or False
DISABLE_DRIVE_LINK = "False" # True or False
USE_SERVICE_ACCOUNTS = "False" # True or False

# qBittorrent/Aria2c
TORRENT_TIMEOUT = "120" # Enter only numbers in seconds
BASE_URL = "http://127.0.0.1" # Enter server IP
BASE_URL_PORT = "80" # Enter server port
WEB_PINCODE = "True" # True or False

# Rclone
RCLONE_PATH = ""
RCLONE_FLAGS = ""
RCLONE_SERVE_URL = ""
RCLONE_SERVE_PORT = "" # Enter only numbers
RCLONE_SERVE_USER = ""
RCLONE_SERVE_PASS = ""

# JDownloader
JD_EMAIL = ""
JD_PASS = ""

# Mega
MEGA_EMAIL = ""
MEGA_PASSWORD = ""

# Sabnzbd
USENET_SERVERS = [{'name': 'main', 'host': '', 'port': 5126, 'timeout': 60, 'username': '', 'password': '', 'connections': 8, 'ssl': 1, 'ssl_verify': 2, 'ssl_ciphers': '', 'enable': 1, 'required': 0, 'optional': 0, 'retention': 0, 'send_group': 0, 'priority': 0}]

# Update
UPSTREAM_REPO = "https://github.com/HuntingBots-Team/TGH_Mirror"
UPSTREAM_BRANCH = "HuntingBots"

# Leech Settings
AS_DOCUMENT = "False" # True or False
LEECH_SPLIT_SIZE = "" # Enter only numbers in bytes
LEECH_FILENAME_PREFIX = ""
LEECH_FILENAME_SUFFIX = ""
LEECH_CAPTION_FONT = "bold" # Options: b, i, u, bi, bu, iu, biu, mono
EQUAL_SPLITS = "False" # True or False
MEDIA_GROUP = "False" # True or False
USER_TRANSMISSION = "True" # True or False
MIXED_LEECH = "True" # True or False
USER_LEECH_DESTINATION = "PM" # PM or DM or chat_id
NAME_SUBSTITUTE = ""
METADATA_TXT = ""
META_ATTACHMENT = ""
THUMBNAIL_LAYOUT = "" # Options: 2x2 or 3x3 or 2x4 or 4x4

# Super Group Settings
REQUEST_LIMITS = "4" # Enter only numbers in seconds
TOKEN_TIMEOUT = "" # Enter only numbers in seconds
MINIMUM_DURATOIN = "" # Enter only numbers in seconds
USER_MAX_TASKS = "" # Enter only numbers
AUTO_DELETE_MESSAGE_DURATION = "120" # Enter only numbers in seconds
ENABLE_MESSAGE_FILTER = "False" # True or False
DELETE_LINKS = "True" # True or False
DM_MODE = "True" # True or False
STOP_DUPLICATE_TASKS = "True" # True or False
INCOMPLETE_TASK_NOTIFIER = "True" # True or False

# Extras
EXTENSION_FILTER = "index html"
YT_DLP_OPTIONS = ""
SET_COMMANDS = "True" # True or False
CMD_SUFFIX = "z"
DEFAULT_UPLOAD = "gd"
STATUS_LIMIT = "5" # Enter only numbers
STATUS_UPDATE_INTERVAL = "11" # Enter only numbers in seconds
DOWNLOAD_DIR = "/usr/src/app/Downloads/"

# Feature Control
DISABLE_BULK = "False" # True or False
DISABLE_MULTI = "False" # True or False
DISABLE_SEED = "False" # True or False
DISABLE_LEECH = "False" # True or False

# Chat IDs
FSUB_IDS = "" # Enter channel id for force subscribe
LOG_CHAT_ID = "-1002251173303" # Enter channel id for task logs
DUMP_CHAT_ID = "-1002367892810" # Enter channel id for leech dump

#Limits
STORAGE_THRESHOLD = "1" # Enter only numbers in GB
TORRENT_LIMIT = "" # Enter only numbers in GB
DIRECT_LIMIT = "" # Enter only numbers in GB
YTDLP_LIMIT = "" # Enter only numbers in GB
PLAYLIST_LIMIT = "" # Enter only numbers
GDRIVE_LIMIT = "" # Enter only numbers in GB
MEGA_LIMIT = "" # Enter only numbers in GB
LEECH_LIMIT = "" # Enter only numbers in GB
CLONE_LIMIT = "" # Enter only numbers in GB
RCLONE_LIMIT = "" # Enter only numbers in GB
JD_LIMIT = "" # Enter only numbers in GB
NZB_LIMIT = "" # Enter only numbers in GB
AVG_SPEED = "" # Enter only numbers in **MB**

#Queueing system
QUEUE_ALL = "10" # Enter only numbers
QUEUE_DOWNLOAD = "" # Enter only numbers
QUEUE_UPLOAD = "" # Enter only numbers

# RSS
RSS_DELAY = "900" # Enter only numbers in seconds
RSS_CHAT = ""

# Torrent Search
SEARCH_API_LINK = ""
SEARCH_LIMIT = "20" # Enter only numbers
SEARCH_PLUGINS = '["https://raw.githubusercontent.com/qbittorrent/search-plugins/master/nova3/engines/piratebay.py",
                   "https://raw.githubusercontent.com/qbittorrent/search-plugins/master/nova3/engines/limetorrents.py",
                   "https://raw.githubusercontent.com/qbittorrent/search-plugins/master/nova3/engines/torlock.py",
                   "https://raw.githubusercontent.com/qbittorrent/search-plugins/master/nova3/engines/torrentscsv.py",
                   "https://raw.githubusercontent.com/qbittorrent/search-plugins/master/nova3/engines/eztv.py",
                   "https://raw.githubusercontent.com/qbittorrent/search-plugins/master/nova3/engines/torrentproject.py",
                   "https://raw.githubusercontent.com/MaurizioRicci/qBittorrent_search_engines/master/kickass_torrent.py",
                   "https://raw.githubusercontent.com/MaurizioRicci/qBittorrent_search_engines/master/yts_am.py",
                   "https://raw.githubusercontent.com/MadeOfMagicAndWires/qBit-plugins/master/engines/linuxtracker.py",
                   "https://raw.githubusercontent.com/MadeOfMagicAndWires/qBit-plugins/master/engines/nyaasi.py",
                   "https://raw.githubusercontent.com/LightDestory/qBittorrent-Search-Plugins/master/src/engines/ettv.py",
                   "https://raw.githubusercontent.com/LightDestory/qBittorrent-Search-Plugins/master/src/engines/glotorrents.py",
                   "https://raw.githubusercontent.com/LightDestory/qBittorrent-Search-Plugins/master/src/engines/thepiratebay.py",
                   "https://raw.githubusercontent.com/v1k45/1337x-qBittorrent-search-plugin/master/leetx.py",
                   "https://raw.githubusercontent.com/nindogo/qbtSearchScripts/master/magnetdl.py",
                   "https://raw.githubusercontent.com/msagca/qbittorrent_plugins/main/uniondht.py",
                   "https://raw.githubusercontent.com/khensolomon/leyts/master/yts.py"]'


config.py
# REQUIRED CONFIG
BOT_TOKEN = "7713486921:AAF4akLKvq-XUZnJg6DBy-nFeRn0GHwa3Hs" # Enter your telegram bot token
OWNER_ID = 6200648859 # Enter your telegram user id
TELEGRAM_API = 3975570 # Enter telegram API key
TELEGRAM_HASH = "680b62f2844aa1954216f6cb99d2f3d9" # Enter telegram API hash

# SEMI-REQUIRED, WE SUGGEST TO FILL IT FROM MONGODB
DATABASE_URL = "mongodb+srv://hunter123:hunter123@cluster0.iueac.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0"

# OPTIONAL CONFIG
TG_PROXY = {}
USER_SESSION_STRING = ""
DOWNLOAD_DIR = "/usr/src/app/downloads/"
CMD_SUFFIX = ""
AUTHORIZED_CHATS = "-1002383407476"
SUDO_USERS = 5623055023, 6200648859
DEFAULT_UPLOAD = "rc"
FILELION_API = ""
STREAMWISH_API = ""
EXCLUDED_EXTENSIONS = ""
INCOMPLETE_TASK_NOTIFIER = False
YT_DLP_OPTIONS = ""
USE_SERVICE_ACCOUNTS = False
NAME_SUBSTITUTE = ""
FFMPEG_CMDS = {}
UPLOAD_PATHS = {}

# INKYPINKY
DELETE_LINKS = False
FSUB_IDS = ""
TOKEN_TIMEOUT = 0
PAID_CHANNEL_ID = 0
PAID_CHANNEL_LINK = ""
SET_COMMANDS = True
METADATA_KEY = ""
WATERMARK_KEY = ""
LOG_CHAT_ID = 0
LEECH_FILENAME_CAPTION = ""
HYDRA_IP = ""
HYDRA_API_KEY = ""
INSTADL_API = ""

# GDrive Tools
GDRIVE_ID = "1-UhbfAlDYGulzPi6UlpVT7pVaZzAutGs"
IS_TEAM_DRIVE = False
STOP_DUPLICATE = False
INDEX_URL = "https://tghthingleech.theghosthunter-tg1.workers.dev/0:"

# Rclone
RCLONE_PATH = ""
RCLONE_FLAGS = ""
RCLONE_SERVE_URL = ""
RCLONE_SERVE_PORT = 0
RCLONE_SERVE_USER = ""
RCLONE_SERVE_PASS = ""

# Mega credentials
MEGA_EMAIL = ""
MEGA_PASSWORD = ""

# Sabnzbd
USENET_SERVERS = [
    {
        "name": "main",
        "host": "",
        "port": 563,
        "timeout": 60,
        "username": "",
        "password": "",
        "connections": 8,
        "ssl": 1,
        "ssl_verify": 2,
        "ssl_ciphers": "",
        "enable": 1,
        "required": 0,
        "optional": 0,
        "retention": 0,
        "send_group": 0,
        "priority": 0,
    },
]

# Update
UPSTREAM_REPO = "https://HuntingBots-Team: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@github.com/HuntingBots-Team/TGH_Mirror"
UPSTREAM_BRANCH = "HuntingBots"

# Leech
LEECH_SPLIT_SIZE = 0
AS_DOCUMENT = False
MEDIA_GROUP = False
USER_TRANSMISSION = False
HYBRID_LEECH = False
LEECH_FILENAME_PREFIX = ""
LEECH_DUMP_CHAT = ""
THUMBNAIL_LAYOUT = ""

# qBittorrent/Aria2c
TORRENT_TIMEOUT = 0
BASE_URL = ""
BASE_URL_PORT = 80
WEB_PINCODE = False

# Queueing system
QUEUE_ALL = 0
QUEUE_DOWNLOAD = 0
QUEUE_UPLOAD = 0

# RSS
RSS_DELAY = 600
RSS_CHAT = ""
RSS_SIZE_LIMIT = 0
